### PR TITLE
fix by_id not free when use_host_ino=true

### PR DESCRIPTION
**Pull request by zyfjeff**
When use_host_ino mode is turned on, we only need to save the by_id relationship if the host inode of the file is larger than MAX_HOST_INO, and don't remove it. however, the previous code used a virtual inode combining the host inode, dev, mnt, and other parts of the file, and this inode is a 56bit inode, which is always larger than MAX_HOST_INO, and so it resulted in the by_id relationship not being cleaned up.

### CODE REVIEW

**Comment by zyfjeff** [comment]
New version of rust clippy bugs
https://github.com/rust-lang/rust-clippy/issues/12560

**Comment by zyfjeff** [comment]
Waiting for this https://github.com/cloud-hypervisor/fuse-backend-rs/pull/187 pr to be merged will solve the CI problem.

File: src/passthrough/mod.rs [hunk 1]
```diff
@@ -777,7 +777,7 @@ impl<S: BitmapSlice + Send + Sync> PassthroughFs<S> {
                         // We just removed the last refcount for this inode.
                         // The allocated inode number should be kept in the map when use_host_ino
                         // is false or inode is bigger than MAX_HOST_INO.
```
Comment (eryugey) [inline src/passthrough/mod.rs:779]: Update the comment as well? "inode" should refer to host inode or real inode?

Comment (zyfjeff) [reply src/passthrough/mod.rs]: fixed

**Review by eryugey — APPROVED** [review]

**Comment by imeoer** [comment]
cc @zyfjeff @eryugey rebasing the master branch will fix the broken CI.

**Pull request merged**